antlr/src/main/antlr/FSH.g4

```diff
@@ -27,6 +29,72 @@ csRule:             concept | caretValueRule | insertRule;
 ruleSet:            KW_RULESET SEQUENCE ruleSetRule+;
 ruleSetRule:        sdRule | concept | vsComponent;
 
+paramRuleSet:       KW_RULESET SEQUENCE PARAMETER_DEF_LIST paramRuleSetRule+;
+paramRuleSetRule:   STAR
```

Comment:
I'm having trouble understanding why this definition has to have the long list of tokens. Would we have to make sure any new tokens that we add are added to this list.

Reply:
in short: yes. The way that the grammar knows that the contents of the `paramRuleSet` are complete is by running into a token that defines the start of a new `entity`. Therefore, this list has to be all tokens that do not start a new `entity`. I tried doing this by negating an alternative of tokens, similar to this:
```
paramRuleSetRule: STAR ~(KW_PROFILE | KW_EXTENSION | KW_INVARIANT | ...)+;
```
But, the resulting parser did not behave as expected.

Reply:
I tried it with a rule like this:
```
paramRuleSetRule:   STAR ~(KW_PROFILE | KW_ALIAS | KW_EXTENSION | KW_INSTANCE | KW_INVARIANT | KW_VALUESET | KW_CODESYSTEM | KW_RULESET | KW_MAPPING)*;
```
And it seemed to work kinda alright. The tests in `FSHImporter.ParamRuleSet.test.ts` fail, but they only fail because for some reason the white space is parsed in a bit differently, and I don't think that would be a huge barrier to overcome. I'm not sure why the `*` works where the `+` doesn't, the difference in logic there isn't totally clicking for me.

If we can make the negation approach work, I think that be a lot less risky going forward if we end up adding new tokens.

Reply:
dang. You are absolutely right. I've updated the grammar accordingly. This is much better. I owe you an ice-cream